File: antigravity/scratch/generate_fixtures.js
import fs from 'fs';
import path from 'path';

const dir = '/Volumes/Black6T/Nexus_Dev/antigravity/test_fixtures';
if (!fs.existsSync(dir)) {
  fs.mkdirSync(dir, { recursive: true });
}

const metadata = '\n［METADATA］\ntags: [test]\n［/METADATA］';

function generateText(chars) {
  const base = 'これはテスト用のダミーテキストです。富士山の上でルビ《ふじさん》を振ったり、会話「おはよう」をしたりします。一行あたりの文字数は適当に調整されます。';
  let text = '';
  while (text.length < chars) {
    text += base + '\n';
  }
  return text.substring(0, chars) + metadata;
}

const fixtures = [
  { name: 'tiny.txt', size: 1000 },
  { name: 'medium.txt', size: 50000 },
  { name: 'large.txt', size: 150000 },
  { name: 'huge.txt', size: 450000 },
];

fixtures.forEach(f => {
  const content = generateText(f.size);
  fs.writeFileSync(path.join(dir, f.name), content);
  console.log(`Created ${f.name} with ${content.length} characters.`);
});

File: antigravity/scratch/generate_fixtures.py
import os

dir_path = '/Volumes/Black6T/Nexus_Dev/antigravity/test_fixtures'
if not os.path.exists(dir_path):
    os.makedirs(dir_path)

metadata = '\n［METADATA］\ntags: [test]\n［/METADATA］'

def generate_text(chars):
    base = 'これはテスト用のダミーテキストです。富士山の上でルビ《ふじさん》を振ったり、会話「おはよう」をしたりします。一行あたりの文字数は適当に調整されます。'
    text = ''
    while len(text) < chars:
        text += base + '\n'
    return text[:chars] + metadata

fixtures = [
    ('tiny.txt', 1000),
    ('medium.txt', 50000),
    ('large.txt', 150000),
    ('huge.txt', 450000),
]

for name, size in fixtures:
    content = generate_text(size)
    with open(os.path.join(dir_path, name), 'w', encoding='utf-8') as f:
        f.write(content)
    print(f'Created {name} with {len(content)} characters.')
